astropy/timeseries/binned.py

```diff
@@ -0,0 +1,277 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+from copy import deepcopy
+
+import numpy as np
+
+from astropy.table import groups, Table, QTable
+from astropy.time import Time, TimeDelta
+from astropy import units as u
+from astropy.units import Quantity
+
+from astropy.timeseries.core import BaseTimeSeries
+
+__all__ = ['BinnedTimeSeries']
+
+
+class BinnedTimeSeries(BaseTimeSeries):
+
+    _require_time_column = False
+
+    def __init__(self, data=None, time_bin_start=None, time_bin_end=None,
```

Comment:
One point for the formalists is that this signature is not consistent with the base `Table` API.  Probably better to copy all the `Table` args in order.  Maybe for 4.0 we can make all args after `data` be keyword only?

Reply:
I agree with making all non-data arguments keyword-only - in fact, I'll do this here since this is a new class, so we might as well do it right straight away. I don't think anyone should rely on positional argument order for ``TimeSeries``.